src/logic/hooks/useEstimateTransactionGas.tsx

```diff
@@ -138,6 +138,21 @@ export const useEstimateTransactionGas = ({
       )
       const isOffChainSignature = checkIfOffChainSignatureIsPossible(isExecution, smartContractWallet, safeVersion)
 
+      if (!isCreation && !isExecution && !approvalAndExecution) {
```

Comment:
How does it work for on-chain approvals? For example with smart contract wallets you have to submit an approval onchain

Reply:
can you just use `isOffChainSignature` flag from above?